### PR TITLE
[Android] Fix Moovit app detection

### PR DESCRIPTION
**Pull request by lorenc-tomasz**
Based on Moovit documentation: https://moovit.com/developers/deeplinking/ to detect if Moovit is installed and can be used we need to check if app is installed instead of checking if URL can be opened. 

PR addresses issue: https://github.com/includable/react-native-map-link/issues/249

### CODE REVIEW

**Comment by tschoffelen** [comment]
I'm not sure if introducing a native dependency is a good idea. This won't work for users using Expo, and makes the package a lot less lightweight.

Any idea why Moovit seems to be the only Android app that we can't detect using the normal method? Are we using the wrong URL scheme? Might be worth trying to extract the Moovit APK and seeing if there is another URL scheme defined in the AndroidManifest there.

**Comment by lorenc-tomasz** [comment]
@tschoffelen Good idea. Will check it tomorrow.

**Comment by lorenc-tomasz** [comment]
@tschoffelen You were right. We can use `moovit://directions`:

```xml

        <activity android:exported="true" android:label="@string/suggest_routes_title" android:launchMode="singleTop" android:name="com.moovit.app.suggestedroutes.SuggestRoutesActivity" android:parentActivityName="com.moovit.app.home.HomeActivity" android:screenOrientation="portrait" android:theme="@style/MoovitTheme.Dark">
            <meta-data android:name="android.support.PARENT_ACTIVITY" android:value=".home.HomeActivity"/>
            <intent-filter>
                <action android:name="android.intent.action.VIEW"/>
                <category android:name="android.intent.category.DEFAULT"/>
                <category android:name="android.intent.category.BROWSABLE"/>
                <data android:host="directions" android:scheme="moovit"/>
            </intent-filter>
            <intent-filter android:label="@string/app_name">
                <action android:name="android.intent.action.VIEW"/>
                <category android:name="android.intent.category.DEFAULT"/>
                <category android:name="android.intent.category.BROWSABLE"/>
                <data android:scheme="geo"/>
            </intent-filter>
        </activity>

```

File: src/constants.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ export function generatePrefixes(options) {
     truckmap: 'truckmap://',
     waze: 'waze://',
     yandex: 'yandexnavi://',
-    moovit: 'moovit://',
+    moovit: isIOS ? 'moovit://' : 'moovit://directions',
```
Comment (tschoffelen) [inline src/constants.js:20]: I think we can include the hostname for both platforms, iOS will still only check the URL scheme.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-native-map-link",
-  "version": "2.11.0",
+  "version": "2.11.1",
```
Comment (tschoffelen) [inline package.json:3]: Mind removing this from the PR? Updating the version number will be done during my release flow.

File: tests/setup.js [hunk 3]
```diff
@@ -1,5 +1,13 @@
-jest.mock('react-native/Libraries/Linking/Linking', () => ({
-  canOpenURL: jest.fn((url) => {
-    return Promise.resolve(url.includes('https://') || url.includes('uber://'));
-  }),
+jest.mock('react-native', () => ({
```
Comment (tschoffelen) [inline tests/setup.js:1]: 👍

**Review by tschoffelen — COMMENTED** [review]
Amazing, thanks for putting in the work to look at this! 👌

Two small suggested changes, once those are fixed I'll merge and release this today!

File: package.json (hunk 2 above)
Comment (lorenc-tomasz) [reply package.json]: Fixed

File: src/constants.js (hunk 1 above)
Comment (lorenc-tomasz) [reply src/constants.js]: Made tests and you are right, we can unify it between platforms. Fixed.

**Comment by lorenc-tomasz** [comment]
Pushed changes

**Review by tschoffelen — APPROVED** [review]

**Comment by tschoffelen** [comment]
Released as `react-native-map-link@2.11.1`

Thanks for your work @lorenc-tomasz!

**Pull request merged**